### PR TITLE
Scaffold compose-debug for Python apps

### PR DESCRIPTION
**Pull request by haniamr**
Instead of volume mapping the debugger, we have it installed as part of the entry-point itself. The debugger is about 5.5 MB so it doesn't slow-up container start.

### CODE REVIEW

File: src/configureWorkspace/configurePython.ts [hunk 1]
```diff
@@ -111,6 +125,20 @@ function genDockerCompose(serviceName: string, ports: number[]): string {
         .replace(/\$ports\$/g, getComposePorts(ports));
 }
 
+function genDockerComposeDebug(serviceName: string, ports: number[], target: PythonTarget, projectType: PythonProjectType): string {
+    const args: string = (inferPythonArgs(projectType, ports) || []).join(' ');
+    const app: string = 'module' in target ? `-m ${target.module}` : target.file;
```
Comment (ravipal) [inline src/configureWorkspace/configurePython.ts:130]: should this also check if `target.module` has some value?

Comment (haniamr) [reply src/configureWorkspace/configurePython.ts:130]: Not really, there will always be a value in either "module" or "file" (we do require non-empty input), the question is which one of them to look at.

**Comment by haniamr** [comment]
> What is the attach story for containers started using docker-compose.yml?

There's a section in the docs about that: https://code.visualstudio.com/docs/containers/debug-python#_how-to-debug-your-app-with-gunicorn

**Comment by karolz-ms** [comment]
> > What is the attach story for containers started using docker-compose.yml?
> 
> There's a section in the docs about that: https://code.visualstudio.com/docs/containers/debug-python#_how-to-debug-your-app-with-gunicorn

And https://code.visualstudio.com/docs/containers/docker-compose#_python

File: src/configureWorkspace/configurePython.ts [hunk 2]
```diff
@@ -60,6 +60,20 @@ services:
...
+const dockerComposeDebugfile = `version: '3.4'
+
+services:
+  $service_name$:
+    image: $service_name$
+    build:
+      context: .
+      dockerfile: Dockerfile
+    entrypoint: /bin/bash
+    command: -c "pip install debugpy -t /tmp && python /tmp/debugpy --wait-for-client --listen 0.0.0.0:5678 $cmd$"
```
Comment (bwateratmsft) [inline src/configureWorkspace/configurePython.ts:72]: How big is `debugpy`? There's an implicit disadvantage that other dependencies are installed via the Dockerfile, and thus benefit from image caching, but this one would not.

Comment (bwateratmsft) [inline src/configureWorkspace/configurePython.ts:72]: Is `--wait-for-client` desirable in this scenario? In contrast, starting a .NET Core app via `docker-compose.debug.yml` will allow the application itself to start and go.

Comment (haniamr) [reply src/configureWorkspace/configurePython.ts:72]: Hmm, that's a valid point. I don't really have a strong opinion, I would have thought that the user might want to debug startup logic as well. Did we ever have specific feedback on this?

Comment (bwateratmsft) [reply src/configureWorkspace/configurePython.ts:72]: Nothing specific, but our answer for debugging startup logic is "F5", not debug-compose.yml.

Comment (haniamr) [reply src/configureWorkspace/configurePython.ts:72]: The problem with F5 is "what if I need more containers to start as well"? Almost every Python example that I saw has at least 1 more container that's usually a database.

Comment (bwateratmsft) [reply src/configureWorkspace/configurePython.ts:72]: Same problem exists for .NET :wink:

**Pull request merged**